aspnet/mvc/views/layout.rst

```diff
@@ -0,0 +1,98 @@
+
+What is a Layout
+----------------
+
+Most web apps have a common layout that provides the user with a consistent experience as they navigate from page to page. The layout typically includes common user interface elements such as the app header, navigation or menu elements, and footer. 
+
+.. image:: layout/_static/page-layout.png
+
+Common HTML structures such as scripts and stylesheets are also frequently used by many pages within an app. All of these shared elements may be defined in a *layout* file, which can then be referenced by any view used within the app. Layouts reduce duplicate code in views, helping them follow the `Don't Repeat Yourself (DRY) principle <http://deviq.com/don-t-repeat-yourself/>`_.
+
+The default Visual Studio ASP.NET MVC project template defines a layout file named ``_Layout.cshtml`` in the ``Views/Shared`` folder:
+
+.. image:: layout/_static/web-project-views.png
+
+This layout defines a top level template for views in the app. Apps do not require a layout, and apps can define more than one layout, with different views specifying different layouts.
+
+An example ``_Layout.cshtml``:
+
+.. literalinclude:: ../../../common/samples/WebApplication1/src/WebApplication1/Views/Shared/_Layout.cshtml
+  :language: html
+  :emphasize-lines: 42,66
+
+Specifying a Layout
+-------------------
+
+Razor views have a ``Layout`` property. Individual views specify a layout by defining this property:
+
+.. literalinclude:: ../../../common/samples/WebApplication1/src/WebApplication1/Views/_ViewStart.cshtml
+  :language: html
+  :emphasize-lines: 2
+
+The layout specified can use a full path (example: ``/Views/Shared/_Layout.cshtml``) or a partial name (example: ``_Layout``). When a partial name is provided, the :doc:`razor view engine <razor>` will search for the layout file using its standard discovery process. That is, the controller-associated folder is searched first, followed by the ``Shared`` folder. This discovery process is identical to the one used to discover :doc:`partial views <partial>`.
+
+Layouts can be nested; each layout can itself specify another layout. When rendering a view, :ref:`ViewStart <viewstart>` is run first, followed by the view itself, followed by its layout(s).
+
+Every layout must call ``RenderBody`` . Wherever the call to ``RenderBody`` is placed, the contents of the view will be rendered.
+
+Sections
+^^^^^^^^
+
+A layout can optionally reference one or more *sections*, by calling ``RenderSection``. Each call to ``RenderSection`` can specify whether that section is required or optional. Individual views specify the content to be rendered within a section using the ``@section`` razor syntax. If a view defines a section, it must be rendered (or an error will occur).
+
+An example ``@section`` definition in a view:
+
+.. literalinclude:: ../../../common/samples/WebApplication1/src/WebApplication1/Views/Manage/VerifyPhoneNumber.cshtml
```

Comment:
Nah, the approach is fine but for a doc trying to explain a section it gives the impression that you run C# in a section when in reality you can have anything there. I'd hate for users to start building partials assuming that's the only way they can include content in a section.

I don't think you need to create a new project here, why not just write the code in place and mark it as HTML? It might not have the perfect syntax highlighting but it'd get the point across.